util/src/main/java/org/datacommons/util/StatVarState.java

```diff
@@ -0,0 +1,67 @@
+package org.datacommons.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.datacommons.proto.Debug;
+import org.datacommons.proto.Mcf;
+
+// Tracks global state on StatVars to help detect whether the same curated DCID is assigned to
+// different StatVars (by content), or if the same StatVar content is assigned different curated
+// DCIDs.  Used by the McfChecker.
+// TODO: Consider expanding this to query DC by generated DCID to find curated DCID in KG.
+public class StatVarState {
+  private final LogWrapper logCtx;
+  private final Map<String, String> generatedToCurated = new HashMap<>();
+  private final Map<String, String> curatedToGenerated = new HashMap<>();
+
+  public StatVarState(LogWrapper logCtx) {
+    this.logCtx = logCtx;
+  }
+
+  boolean check(String id, Mcf.McfGraph.PropertyValues node) {
+    var curatedDcid = McfUtil.getPropVal(node, Vocabulary.DCID);
+    if (curatedDcid.isEmpty()) {
+      // This will have been handled in the McfChecker.
+      return false;
+    }
+
+    var generated = DcidGenerator.forStatVar(id, node);
+    if (generated.dcid.isEmpty()) {
+      // This is due to malformed SV node, which should have been handled in the checker.
+      return false;
+    }
+
+    var existingGeneratedDcid = curatedToGenerated.getOrDefault(curatedDcid, null);
+    if (existingGeneratedDcid != null && !existingGeneratedDcid.equals(generated.dcid)) {
+      logCtx.addEntry(
+          Debug.Log.Level.LEVEL_ERROR,
+          "Checker_SameDcidForDifferentStatVars",
```

Comment:
the prefix "Checker_" seems a little ambiguous as the counter key. Maybe this could also take on the "MCF_" prefix or the "Sanity_" prefix. Same for next log entry.

Reply:
agreed... went with Sanity_ since this gets called in the context of the McfChecker which uses that.